### ISSUE TITLE
Sync multisecret from json

### ISSUE THREAD

**Issue by kubermihe** [SparebankenVest/azure-key-vault-to-kubernetes#416]
Hi,

at the moment i'm managing my argocd helm-repositories in azure as secret and sync them with akv2k8s as a Multi-Key-Value secret (https://akv2k8s.io/tutorials/sync/4-multi-key-value-secret/). The procedure is great and everything is fine with that. Here's one sample:

```
{
  "name": "test",
  "url": "https://path.to.my.nexus.org/repository/test/",
  "type": "helm",
  "username": "test-user",
  "password": "test-password"
}
```

To hold all the repositories in one secret (just to keep them together), I've tried to create a json with multiple arrays. My goal is to hold one secret in Azure and Split it array by array into single kubernetes secrets. Now it's possible to sync multiple Azure secrets into one kubernetes-secret(https://akv2k8s.io/tutorials/sync/5-multi-akvs-to-one-secret/), but not like first described.

Do i have some syntax error, or isn't it possible at the moment to sync it like that?

```
{
  "repo1": {
    "name": "repo1",
    "url": "https://path.to.my.nexus.org/repository/repo1/",
    "type": "helm",
    "username": "test-user",
    "password": "test-password"
  }
  "repo2": {
    "name": "repo2",
    "url": "https://path.to.my.nexus.org/repository/repo2/",
    "type": "helm",
    "username": "test-user",
    "password": "test-password"
  }
}
```

Please let me know if my understanding is completely wrong or if there's already something in planing.

**Comment by kubermihe** [comment]
Hello @tspearconquest ,

yes i've already tried the nested example with link 4, and the result is that all information inside the azure secret is synced into one secret. Our goal is to hold all the information in one secret and split the several array-clusters into different secrets.

The information is synced for argocd, but argocd needs one k8s-secret per repository. Our goal is to bring all repositories into the azure secret and split it in k8s, where argocd is able to scrape the information from several secrets.

**Comment by tspearconquest** [comment]
Ah, so one keyvault secret to multiple kubernetes secrets. Sort of the reverse of what the tutorials demonstrate, right?

**Comment by kubermihe** [comment]
Yes, but not split with the same values. Each array has to be extracted to its own k8s-secret.

**Comment by tspearconquest** [comment]
Understood, thanks for clarifying. This isn't currently supported. There would need to be some additional work on the CRD to support this, as the user (you, in this case) would need to provide something in the `kind: AzureKeyVaultSecret` resource to tell akv2k8s which array pulled from the keyvault secret should be synced with which kubernetes secret.

**Comment by kubermihe** [comment]
Yes that's correct. Do you think this feature-request is realistic and you can implement this? Or is it not implementable, because it's not in the scope of azure to store various secrets in one and split it in k8s?

**Comment by tspearconquest** [comment]
I do think its realistic, however I'm simply another user, and not very familiar with Golang. So I couldn't implement it myself. @181192 and @torresdal are maintainers and could advise on whether this is something the team is willing to implement in the future.
